components/nimbus/src/stateful/targeting.rs

```diff
@@ -0,0 +1,31 @@
+use crate::{
+    enrollment::ExperimentEnrollment, stateful::behavior::EventStore, NimbusTargetingHelper,
+    TargetingAttributes,
+};
+use std::sync::{Arc, Mutex};
+
+impl NimbusTargetingHelper {
+    pub(crate) fn with_targeting_attributes(
+        targeting_attributes: &TargetingAttributes,
+        event_store: Arc<Mutex<EventStore>>,
+    ) -> Self {
+        Self {
+            context: serde_json::to_value(targeting_attributes.clone()).unwrap(),
+            event_store,
+            targeting_attributes: Arc::new(Mutex::new(Some(targeting_attributes.clone()))),
+        }
+    }
+
+    pub(crate) fn update_enrollment(&mut self, enrollment: ExperimentEnrollment) -> bool {
+        let mut ta = self.targeting_attributes.lock().unwrap();
+        if ta.is_some() {
```

Comment:
We don't need to unwrap() here: we can use an if-let statement.

```suggestion
        if let Some(mut attrs) = self.targeting_attributes.lock().expect("...") {
```